beacon-chain/core/validators/validator.go

```diff
@@ -440,6 +441,45 @@ func CheckValidatorMinDeposit(validatorSet []*pb.ValidatorRecord, currentSlot ui
 	return validatorSet
 }
 
+// NewRegistryDeltaChainTip returns the new validator registry delta chain tip.
+//
+// Spec pseudocode definition:
+//   def get_new_validator_registry_delta_chain_tip(current_validator_registry_delta_chain_tip: Hash32,
+//                                               validator_index: int,
+//                                               pubkey: int,
+//                                               flag: int) -> Hash32:
+// 	  """
+//    Compute the next root in the validator registry delta chain.
+//    """
+//    return hash_tree_root(
+//        ValidatorRegistryDeltaBlock(
+//            latest_registry_delta_root=current_validator_registry_delta_chain_tip,
+//            validator_index=validator_index,
+//            pubkey=pubkey,
+//            flag=flag,
+//        )
+//    )
+func NewRegistryDeltaChainTip(
+	flag uint64,
+	index uint32,
+	pubKey []byte,
+	currentValidatorRegistryDeltaChainTip []byte) ([32]byte, error) {
+
+	newDeltaChainTip := &pb.ValidatorRegistryDeltaBlock{
+		LatestRegistryDeltaRootHash32: currentValidatorRegistryDeltaChainTip,
+		ValidatorIndex:                index,
+		Pubkey:                        pubKey,
+		Flag:                          flag,
+	}
+
+	// TODO(716): Replace serialization with tree hash function.
+	serializedChainTip, err := proto.Marshal(newDeltaChainTip)
```

Comment:
I have been seeing this pattern all over the repo with our proto objects. Imo we should not be having this for hashing, Ideally we have a generalized hashing function which can take in any object that fulfills the proto interface and then use that to hash the object. The function should be in hashutils and removed from our core package

Reply:
Will we even be tree-hasing the proto object? If the other clients don't use the same proto object then we'll never reach consensus. Ideally, we used be using the SSZ state object then apply tree hash on top of that. That's something to work on next

Reply:
Probably not , with jie hou tackling this now we will be using tree hashing mostly. tree hashing is used majorly across the spec, so that is what we will be primarily using. However for the places where we need to keccak hash, we should have a generalized function for it. Instead of implementing a separate hashing function for each proto primitive.

Reply:
but we won't be hashing the proto field right? we'll be hashing struct. Here is the hashing function https://github.com/prysmaticlabs/prysm/blob/master/shared/hashutil/hash.go#L10

Reply:
I was referring to something like this 
```go
type proto interface {
 Marshal() []byte
}

func HashProto(object proto) [] byte {
 marshalledObj := object.Marshal()
 return Hash(marshalledObj)
}
```
We could use it to abstract the marshalling and then hashing everytime.